contrib_bots/lib/hangman.py

```diff
@@ -0,0 +1,170 @@
+    elif split[2] in bot_self.guessed:
+        bot_self.output = "You have already guessed the letter " \
+            "" + split[2] + "! To guess again, type @hangman guess <letter>."
+    elif split[2] not in "abcdefghijklmnopqrstuvwxyz":
+        bot_self.output = split[2] + " is not a letter! To guess a letter, " \
+            "type @hangman guess <letter>."
+    else:
+        bot_self.guessed.append(split[2])
+        if split[2] not in bot_self.currentword:
+            bot_self.stage += 1
+            if bot_self.stage == 10:
+                end = "\n·\n:crying_cat_face: Oh no, you lost! " \
+                    "The correct word was " + str(bot_self.currentword) + "." \
+                      " To play again, type @hangman new."
+                bot_self.ended = True
+        stages = ["", "```\n=============\n```",
+                  ("```\n           |\n           |\n           |" +
+                   "\n           |\n           |\n=============\n```"),
+                  ("```\n   +-------+\n           |\n           |\n    " +
+                   "       |\n           |\n           |\n=============\n```"),
+                  ("```\n   +-------+\n   |       |\n           |\n    " +
+                   "       |\n           |\n           |\n=============\n```"),
+                  ("```\n   +-------+\n   |       |\n   0       |\n    " +
+                   "       |\n           |\n           |\n=============\n```"),
+                  ("```\n   +-------+\n   |       |\n   0       |\n   |" +
+                   "       |\n           |\n           |\n=============\n```"),
+                  ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                   "       |\n           |\n           |\n=============\n```"),
+                  ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                   "\      |\n           |\n           |\n=============\n```"),
+                  ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                   "\      |\n  /        |\n           |\n=============\n```"),
+                  ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                   "\      |\n  / \      |\n           |\n=============\n```")]
+        outcur = []
+        wrongguess = []
+        for i in range(0, len(bot_self.currentword)):
+            if bot_self.currentword[i] in bot_self.guessed:
+                outcur.append(str(bot_self.currentword[i]))
+            else:
+                outcur.append("_")
+        if "_" not in outcur:
+            end = "\n·\n:party_popper: Congratulations! You won! " \
+                "To play again, type @hangman new. :party_popper:"
+            bot_self.ended = True
+        for i in range(0, len(bot_self.guessed)):
+            if bot_self.guessed[i] not in bot_self.currentword:
+                wrongguess.append(str(bot_self.guessed[i]))
+        bot_self.output = (stages[bot_self.stage] + "\n·\n```" +
```

Comment:
`alt+space`? When I press alt and space at the same time it opens a little menu at the top-right of my browser.